packages/enzyme-test-suite/test/shared/hooks/useContext.jsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { expect } from 'chai';
+import wrap from 'mocha-wrap';
+import sinon from 'sinon-sandbox';
+import { Portal } from 'react-is';
+
+import { render } from 'enzyme';
+import getAdapter from 'enzyme/build/getAdapter';
+
+import {
+  describeIf,
+  itIf,
+} from '../../_helpers';
+import {
+  is,
+  REACT16,
+} from '../../_helpers/version';
+
+import {
+  useContext,
+  useState,
+  createContext,
+} from '../../_helpers/react-compat';
+
+export default function describeUseContext({
+  hasHooks,
+  Wrap,
+  WrapRendered,
+  Wrapper,
+  WrapperName,
+  isShallow,
+  isMount,
+  makeDOMElement,
+}) {
+  describeIf(hasHooks, 'hooks: useContext', () => {
+    describe('simple example', () => {
+      let TitleContext;
+      beforeEach(() => {
+        TitleContext = createContext();
+      });
+
+      const UiComponent = props => {
+        let title = useContext(TitleContext);
+        return (
+          <div>
+            <b>UiComponent: </b>
+            {title}
+          </div>
+        );
+      };
+
+      const contextTitle = 'CustomTitle';
+
+      const App = () => {
+        return (
+          <TitleContext.Provider value={contextTitle}>
+            <UiComponent />
+          </TitleContext.Provider>
+        );
+      };
+
+      // TODO; useContext seems to not work in the shallow renderer's initial render
+      itIf(!isShallow, 'renders App with TitleContext', () => {
+        const wrapper = Wrap(<App />);
+        wrapper.update();
+        const subWrapper = isShallow ? wrapper.dive().dive() : wrapper;
+        expect(subWrapper.text()).to.equal(`UiComponent: ${contextTitle}`);
+      });
+    });
+
+    // TODO: useContext: enable when shallow dive supports createContext
```

Comment:
and also add the link here

Reply:
See https://github.com/airbnb/enzyme/pull/2008#discussion_r292263333